Milestone4/OfflineStudentManagementSystem/Control/services/staffServices.ts

```diff
@@ -0,0 +1,184 @@
+import { instantStudent, staffMenu} from '../../View/app';
+import { Teacher } from '../../Model/factoryComponents/teacher';
+import { menu, staffOperations } from '../../View/app';
+import { validateStaffLogin, checkIsExist, validateuserDetails, readLine , validateRollNo, errorReport } from '../util/utils'
+
+export const questionAsync = (prompt: string): Promise<string> => {
+  return new Promise((res) => {
+    readLine.question(prompt, (answer: string) => {
+      res(answer);
+    });
+  });
+};
+
+export const staffLogin = async () => {
+  try {
+    const type = 'teacher';
+    const name = await questionAsync('Enter staff name: ');
+    const password = await questionAsync('Enter staff password: ');
+    validateStaffLogin(name, password);
+    Teacher.instanceOfTeacher.isActive = true;
+    if (Teacher.instanceOfTeacher.isActive) {
+      staffOperations();
+    } else {
+      throw new Error('Not logged In')
+    }
+    return staffMenu();
+  } catch (err) {
+    let message;
+    if (err instanceof Error) {
+      message = err.message;
+    } else {
+      message = String(message);
+    }
+    reportError({ message })
+  }
+};
+
+export const addStudentDetails = async () => {
+  try {
+    if (Teacher.instanceOfTeacher.isActive) {
+      const getRollNo = Number(
+        await questionAsync('Enter Roll No to add details : ')
+      );
+      const isExist = checkIsExist(getRollNo);
+      if (await isExist) {
```

Comment:
Is it a proper method to use await inside if condition

Reply:
fixed